src/main/java/com/applicaster/sport1player/Sport1PlayerAdapter.java

```diff
@@ -129,11 +129,11 @@ protected void openLoginPluginIfNeeded(final boolean isInline) {
         final LoginContract loginPlugin = LoginManager.getLoginPlugin();
         this.isInline = isInline;
         if (loginPlugin != null ){
-            accessToken = loginPlugin.getToken();
             loginPlugin.isItemLocked(getContext(), getFirstPlayable(), result -> {
                 if (result) {
                     loginPlugin.login(getContext(), getFirstPlayable(), null, loginResult -> {
                         if (loginResult) {
+                            accessToken = loginPlugin.getToken();
```

Comment:
Getting the token here you can have a situation where the token is not set because the user has alerady logged in.
I would put before the tokenedUrl call.

Reply:
updated